packages/flutter/lib/src/material/chip.dart

```diff
@@ -2049,7 +2051,9 @@ class _RenderChip extends RenderBox with SlottedContainerRenderObjectMixin<_Chip
     final Color disabledColor = _disabledColor;
     final int disabledColorAlpha = disabledColor.alpha;
     if (needsCompositing) {
-      context.pushLayer(OpacityLayer(alpha: disabledColorAlpha), paintWithOverlay, offset);
+      final OpacityLayer newLayer = OpacityLayer(alpha: disabledColorAlpha);
+      _layerHandle.layer = newLayer;
+      context.pushLayer(newLayer, paintWithOverlay, offset);
     } else {
       if (disabledColorAlpha != 0xff) {
```

Comment:
Looks like the same problem exists below in line 2078.

Reply:
do you mean `_paintAvatar`?

After I applied similar fix, I got failures of chip_test.dart:

```
The following assertion was thrown during paint():
Attempted to create a handle to an already disposed layer: OpacityLayer#ae49c(handles: 0, offset:
Offset(322.0, 284.0), alpha: 255).
'package:flutter/src/rendering/layer.dart':
Failed assertion: line 786 pos 7: 'layer?.debugDisposed != true'

Either the assertion indicates an error in the framework itself, or we should provide substantially
more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.yml

When the exception was thrown, this was the stack:
#2      LayerHandle.layer= (package:flutter/src/rendering/layer.dart:786:7)
#3      ContainerLayer.append (package:flutter/src/rendering/layer.dart:1251:25)
#4      PaintingContext.appendLayer (package:flutter/src/rendering/object.dart:292:21)
#5      PaintingContext.pushLayer (package:flutter/src/rendering/object.dart:483:5)
#6      PaintingContext.pushOpacity (package:flutter/src/rendering/object.dart:716:5)
#7      _RenderChip._paintAvatar (package:flutter/src/material/chip.dart:2057:51)
#8      _RenderChip.paint (package:flutter/src/material/chip.dart:2112:5)
#9      RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:3210:7)
#10     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:250:13)
```

Reply:
moved the change for _paintChild to a separate PR: https://github.com/flutter/flutter/pull/135228

Reply:
That exception is likely caused by the problem I describe in my comment on line 2043/2044.